aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -629,6 +654,8 @@ public class AwsProperties implements Serializable {
   public static final String LAKE_FORMATION_DB_NAME = "lakeformation.db-name";
 
   private static final String HTTP_CLIENT_PREFIX = "http-client.";
+
+  private static final String CREDENTIAL_PROVIDER_PREFIX = "client.credentials-provider.";
```

Comment:
nit: newline after class variables before instance variables

Reply:
Shall we add some description here or above(since this is a private String) to show that users can use 
```java
client.credentials-provider.my_key=my_value
```
to pass additional properties to the credentials provider?

like:https://github.com/apache/iceberg/blob/c92efd46515f5352f4a6e1dd37ead9cfb6e86a9d/aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java#L603-L611